### PR TITLE
Add htmx-lsp support

### PR DESCRIPTION
**Pull request by OZoneGuy**

### CODE REVIEW

File: lua/lspconfig/server_configurations/htmx.lua [hunk 1]
```diff
@@ -0,0 +1,27 @@
+local util = require 'lspconfig.util'
+-- NOTE: the binary can be installed in custom a location. The user should
+-- change the cmd to match the installation path
+local cargo_home = os.getenv 'CARGO_INSTALL_ROOT' or os.getenv 'CARGO_HOME' or os.getenv 'HOME' .. '/.cargo'
+
+return {
+  default_config = {
+    cmd = { cargo_home .. '/bin/htmx-lsp' },
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/htmx.lua:8]: This doesn't make sense. If I install the binary somewhere else. will fail here

Comment (OZoneGuy) [reply lua/lspconfig/server_configurations/htmx.lua]: That is correct. Pushed a change to call without the path.

File: lua/lspconfig/server_configurations/htmx.lua [hunk 2]
```diff
@@ -0,0 +1,24 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = { 'htmx-lsp' },
+    filetypes = { 'html' },
+    root_dir = function(fname)
+      return util.find_git_ancestor(fname) or vim.fn.expand '%:p:h' or vim.loop.os_homedir()
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/htmx.lua:8]:
```suggestion
      return util.find_git_ancestor(fname)
```

File: lua/lspconfig/server_configurations/htmx.lua [hunk 3]
```diff
@@ -0,0 +1,24 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = { 'htmx-lsp' },
+    filetypes = { 'html' },
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/htmx.lua:6]:
```suggestion
    filetypes = { 'html' },
    single_file_support = true,
```

**Review by glepnir — CHANGES_REQUESTED** [review]

File: lua/lspconfig/server_configurations/htmx.lua [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    single_file_support = true,
+    root_dir = function(fname)
+      return util.find_git_ancestor(fname)
+    end,
+    settings = {},
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/htmx.lua:11]: no need

**Comment by OZoneGuy** [comment]
Thanks for the review and your time! Really appreciate it.

**Pull request merged**